smack-websocket/src/main/java/org/jivesoftware/smack/websocket/rce/WebsocketRemoteConnectionEndpoint.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket.rce;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jivesoftware.smack.datatypes.UInt16;
+import org.jivesoftware.smack.util.rce.RemoteConnectionEndpoint;
+
+public final class WebsocketRemoteConnectionEndpoint implements RemoteConnectionEndpoint {
+
+    private static final Logger LOGGER = Logger.getAnonymousLogger();
+
+    private final String protocol;
+    private final String host;
+    private final int port;
+    private String description;
+
+    public WebsocketRemoteConnectionEndpoint(String href) {
```

Comment:
I assume the right type for `href` would be `URL` not `String`. That should make your live easier. You may also want to keep `WeboscketRemoteConnectionEndpoint(String)` constructor, which tries to transform the string to an URL and calls the right constructor.